js/src/admin/components/AdminNav.js

```diff
@@ -31,76 +31,119 @@ export default class AdminNav extends Component {
         },
         app.translator.trans('core.admin.nav.basics_button')
       )
     );
 
     items.add(
       'mail',
-      AdminLinkButton.component(
+      LinkButton.component(
         {
           href: app.route('mail'),
           icon: 'fas fa-envelope',
-          description: app.translator.trans('core.admin.nav.email_text'),
+          title: app.translator.trans('core.admin.nav.email_title'),
         },
         app.translator.trans('core.admin.nav.email_button')
       )
     );
 
     items.add(
       'permissions',
-      AdminLinkButton.component(
+      LinkButton.component(
         {
           href: app.route('permissions'),
           icon: 'fas fa-key',
-          description: app.translator.trans('core.admin.nav.permissions_text'),
+          title: app.translator.trans('core.admin.nav.permissions_title'),
         },
         app.translator.trans('core.admin.nav.permissions_button')
       )
     );
 
     items.add(
       'appearance',
-      AdminLinkButton.component(
+      LinkButton.component(
         {
           href: app.route('appearance'),
           icon: 'fas fa-paint-brush',
-          description: app.translator.trans('core.admin.nav.appearance_text'),
+          title: app.translator.trans('core.admin.nav.appearance_title'),
         },
         app.translator.trans('core.admin.nav.appearance_button')
       )
     );
 
     items.add(
-      'extensions',
-      AdminLinkButton.component(
-        {
-          href: app.route('extensions'),
-          icon: 'fas fa-puzzle-piece',
-          description: app.translator.trans('core.admin.nav.extensions_text'),
-        },
-        app.translator.trans('core.admin.nav.extensions_button')
-      )
+      'search',
+      <div className="Search-input">
+        <input
+          className="FormControl SearchBar"
+          bidi={this.query}
+          type="search"
+          placeholder={app.translator.trans('core.admin.nav.search_placeholder')}
+        />
+      </div>
     );
 
     return items;
   }
+
+  extensionItems() {
+    const items = new ItemList();
+
+    const categorizedExtensions = getCategorizedExtensions();
+    const categories = app.extensionCategories;
+
+    Object.keys(categorizedExtensions).map((category) => {
+      if (!this.query()) {
+        items.add(
+          `${category} NavDivider`,
+          <h4 className="ExtensionListTitle">{app.translator.trans(`core.admin.nav.categories.${category}`)}</h4>,
+          categories[category]
+        );
+      }
+
+      categorizedExtensions[category].map((extension) => {
+        const query = this.query().toUpperCase();
+
+        if (
+          !query ||
+          extension.extra['flarum-extension'].title.toUpperCase().includes(query) ||
+          extension.description.toUpperCase().includes(query)
+        ) {
+          items.add(
+            `${extension.id} ExtensionItem`,
+            ExtensionLinkButton.component(
```

Comment:
I would convert this to JSX - we want to avoid the .component() stuff as much as possible, and having JSX makes it more readable.